sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/CombiningLimitsSuite.scala

```diff
@@ -117,4 +130,85 @@ class CombiningLimitsSuite extends PlanTest {
       testRelation.select().groupBy()(count(1)).orderBy(count(1).asc).analyze)
     comparePlans(optimized4, expected4)
   }
+
+  test("SPARK-33497: Override maxRows in some LogicalPlan") {
+    def checkPlan(p1: LogicalPlan, p2: LogicalPlan): Unit = {
+      comparePlans(Optimize.execute(p1.analyze), p2.analyze)
+    }
+
+    // test LocalRelation
+    checkPlan(
+      testRelation.select().limit(10),
+      testRelation.select()
+    )
+
+    // test Range
+    checkPlan(
+      Range(1, 100, 1, None).select().limit(200),
+      Range(1, 100, 1, None).select()
+    )
+
+    // test Sample
+    val sampleQuery = testRelation.select().sample(upperBound = 0.2).limit(10).analyze
+    val sampleOptimized = Optimize.execute(sampleQuery)
+    assert(sampleOptimized.collect { case l @ Limit(_, _) => l }.isEmpty)
+
+    // test Deduplicate
+    checkPlan(
+      testRelation.deduplicate(Symbol("a")).limit(10),
```

Comment:
nit: `'a`

Reply:
@cloud-fan Is these case needed because of [SPARK-29392](https://issues.apache.org/jira/browse/SPARK-29392) ?

Reply:
ah I see. It's weird that the previous code still uses `'a`... Anyway let's use Symbol now.